fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/transactionprocessor/impl/ChargeOrTransaction.java

```diff
@@ -58,7 +58,7 @@ private LocalDate getEffectiveDate() {
     }
 
     private boolean isBackdatedCharge() {
-        return loanCharge.get().getDueDate().isBefore(loanCharge.get().getSubmittedOnDate());
+        return (loanCharge.get().getDueDate() != null && loanCharge.get().getDueDate().isBefore(loanCharge.get().getSubmittedOnDate()));
```

Comment:
Unnecessary change. Would you mind to revert? If it is a fix for a bug ticket, please extract it into a new PR!

Reply:
Nothing changed here either.... :/